lib/config.js

```diff
@@ -501,9 +501,13 @@ AWS.Config = AWS.util.inherit({
    */
   setPromisesDependency: function setPromisesDependency(dep) {
     PromisesDependency = dep;
+    // if null was passed in, we should try to use native promises
+    if (dep === null && typeof Promise === 'function') {
```

Comment:
This should be reflected in the jsdoc comment above: "... If native promises are supported, pass in `null` to use direct the SDK to use them. If native promises are not supported and `null` is provided for this argument, the method is a no-op."